pootle/static/js/editor/containers/Editor.js

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) Pootle contributors.
+ *
+ * This file is a part of the Pootle project. It is distributed under the GPL3
+ * or later license. See the LICENSE file for a copy of the license and the
+ * AUTHORS file for copyright and authorship information.
+ */
+
+import React from 'react';
+
+import EditorTextarea from '../components/EditorTextarea';
+import { getAreaId } from '../utils';
+
+
+const Editor = React.createClass({
+
+  propTypes: {
+    initialValues: React.PropTypes.array,
+    isRawMode: React.PropTypes.bool,
+    locale: React.PropTypes.string,
+    localeDir: React.PropTypes.string,
+    // FIXME: needed to allow interaction from the outside world. Remove ASAP.
+    onChange: React.PropTypes.func.isRequired,
+    style: React.PropTypes.object,
+    targetNplurals: React.PropTypes.number.isRequired,
+    values: React.PropTypes.array,
+  },
+
+  childContextTypes: {
+    locale: React.PropTypes.string,
+    localeDir: React.PropTypes.string,
+  },
+
+  getChildContext() {
+    return {
+      locale: this.props.locale,
+      localeDir: this.props.localeDir,
+    };
+  },
+
+  render() {
+    const editorTextareas = [];
+    const { values } = this.props;
+
+    for (let i=0; i < this.props.targetNplurals; i++) {
+      const extraProps = {};
+      if (values !== undefined) {
+        extraProps.value = values[i];
+        // XXX: hacky way to denote the value was already passed down
+        values[i] = null;
+      }
+      if (this.props.isRawMode !== undefined) {
+        extraProps.isRawMode = this.props.isRawMode;
+      }
+      editorTextareas.push(
+        <EditorTextarea
+          id={getAreaId(i)}
+          initialValue={this.props.initialValues[i]}
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Expected an assignment or function call and instead saw an expression.](https://www.codacy.com/app/dwayne/pootle/file/2100760359/issues/source?bid=1990595&fileBranchId=2945338#l58)